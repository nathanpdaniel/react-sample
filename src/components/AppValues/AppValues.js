import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './AppValues.scss';

export const AppValues = ({ onBack, values: { sampleSize, sampleMean, standardDeviation, testHypothesis, hypothesizedMean }}) => (
  <div className="appValues">
    <div>
      <div className="label">Sample Size:</div>
      <div>{sampleSize}</div>
    </div>
    <div>
      <div className="label">Standard Mean:</div>
      <div>{sampleMean}</div>
    </div>
    <div>
      <div className="label">Standard Deviation:</div>
      <div>{standardDeviation}</div>
    </div>
    {testHypothesis &&
    <div>
      <div className="label">Hypothesized Mean</div>
      <div>{hypothesizedMean}</div>
    </div>
    }
    <div className={classnames('buttons')}>
      <button onClick={onBack}>Back</button>
    </div>
  </div>
);

AppValues.propTypes = {
  values: PropTypes.shape({
    sampleSize: PropTypes.number.isRequired,
    sampleMean: PropTypes.number.isRequired,
    standardDeviation: PropTypes.number.isRequired,
    testHypothesis: PropTypes.bool.isRequired,
    hypothesizedMean: PropTypes.number
  }),
  onBack: PropTypes.func.isRequired,
}
