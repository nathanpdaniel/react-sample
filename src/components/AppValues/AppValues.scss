.appValues {
  display: flex;
  flex-direction: column;
  padding: 24px;

  > div {
    display: flex;
    width: 100%;

    div {
      padding: 12px;
      border: 1px solid lightgray;
      width: 100%;

      &.label {
        width: 25%;
        font-size: 12px;
      }
    }

    &.buttons {
      padding: 12px 0;
      display: flex;
      justify-content: flex-start;

      button {
        border-radius: 4px;
        margin: 0 8px;
        min-width: 80px;
        padding: 10px 20px;
      }
    }
  }
}