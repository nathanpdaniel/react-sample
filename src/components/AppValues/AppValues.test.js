import React from 'react';
import { shallow } from 'enzyme';
import { AppValues } from './AppValues';

describe('AppValues Tests', () => {
  it('should display properly sans test hypothesis', () => {
    const onBack = jest.fn();
    const values = {
      sampleSize: 123,
      sampleMean: 1.23,
      standardDeviation: -1.5,
      testHypothesis: false
    }
    const component = shallow(<AppValues values={values} onBack={onBack} />);
    expect(component).toMatchSnapshot();
  });
  it('should display properly with test hypothesis', () => {
    const onBack = jest.fn();
    const values = {
      sampleSize: 123,
      sampleMean: 1.23,
      standardDeviation: -1.5,
      testHypothesis: true,
      hypothesizedMean: 2.34
    }
    const component = shallow(<AppValues values={values} onBack={onBack} />);
    expect(component).toMatchSnapshot();
  });
});