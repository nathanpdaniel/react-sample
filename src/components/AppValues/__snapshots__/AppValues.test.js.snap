// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppValues Tests should display properly sans test hypothesis 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppValues
    onBack={[MockFunction]}
    values={
      Object {
        "sampleMean": 1.23,
        "sampleSize": 123,
        "standardDeviation": -1.5,
        "testHypothesis": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <div
            className="label"
          >
            Sample Size:
          </div>
          <div>
            123
          </div>
        </div>,
        <div>
          <div
            className="label"
          >
            Standard Mean:
          </div>
          <div>
            1.23
          </div>
        </div>,
        <div>
          <div
            className="label"
          >
            Standard Deviation:
          </div>
          <div>
            -1.5
          </div>
        </div>,
        false,
        <div
          className="buttons"
        >
          <button
            onClick={[MockFunction]}
          >
            Back
          </button>
        </div>,
      ],
      "className": "appValues",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              Sample Size:
            </div>,
            <div>
              123
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sample Size:",
              "className": "label",
            },
            "ref": null,
            "rendered": "Sample Size:",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 123,
            },
            "ref": null,
            "rendered": 123,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              Standard Mean:
            </div>,
            <div>
              1.23
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Standard Mean:",
              "className": "label",
            },
            "ref": null,
            "rendered": "Standard Mean:",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1.23,
            },
            "ref": null,
            "rendered": 1.23,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              Standard Deviation:
            </div>,
            <div>
              -1.5
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Standard Deviation:",
              "className": "label",
            },
            "ref": null,
            "rendered": "Standard Deviation:",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": -1.5,
            },
            "ref": null,
            "rendered": -1.5,
            "type": "div",
          },
        ],
        "type": "div",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            onClick={[MockFunction]}
          >
            Back
          </button>,
          "className": "buttons",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Back",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <div
              className="label"
            >
              Sample Size:
            </div>
            <div>
              123
            </div>
          </div>,
          <div>
            <div
              className="label"
            >
              Standard Mean:
            </div>
            <div>
              1.23
            </div>
          </div>,
          <div>
            <div
              className="label"
            >
              Standard Deviation:
            </div>
            <div>
              -1.5
            </div>
          </div>,
          false,
          <div
            className="buttons"
          >
            <button
              onClick={[MockFunction]}
            >
              Back
            </button>
          </div>,
        ],
        "className": "appValues",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                Sample Size:
              </div>,
              <div>
                123
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sample Size:",
                "className": "label",
              },
              "ref": null,
              "rendered": "Sample Size:",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 123,
              },
              "ref": null,
              "rendered": 123,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                Standard Mean:
              </div>,
              <div>
                1.23
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Standard Mean:",
                "className": "label",
              },
              "ref": null,
              "rendered": "Standard Mean:",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1.23,
              },
              "ref": null,
              "rendered": 1.23,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                Standard Deviation:
              </div>,
              <div>
                -1.5
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Standard Deviation:",
                "className": "label",
              },
              "ref": null,
              "rendered": "Standard Deviation:",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": -1.5,
              },
              "ref": null,
              "rendered": -1.5,
              "type": "div",
            },
          ],
          "type": "div",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[MockFunction]}
            >
              Back
            </button>,
            "className": "buttons",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Back",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Back",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`AppValues Tests should display properly with test hypothesis 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppValues
    onBack={[MockFunction]}
    values={
      Object {
        "hypothesizedMean": 2.34,
        "sampleMean": 1.23,
        "sampleSize": 123,
        "standardDeviation": -1.5,
        "testHypothesis": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <div
            className="label"
          >
            Sample Size:
          </div>
          <div>
            123
          </div>
        </div>,
        <div>
          <div
            className="label"
          >
            Standard Mean:
          </div>
          <div>
            1.23
          </div>
        </div>,
        <div>
          <div
            className="label"
          >
            Standard Deviation:
          </div>
          <div>
            -1.5
          </div>
        </div>,
        <div>
          <div
            className="label"
          >
            Hypothesized Mean
          </div>
          <div>
            2.34
          </div>
        </div>,
        <div
          className="buttons"
        >
          <button
            onClick={[MockFunction]}
          >
            Back
          </button>
        </div>,
      ],
      "className": "appValues",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              Sample Size:
            </div>,
            <div>
              123
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sample Size:",
              "className": "label",
            },
            "ref": null,
            "rendered": "Sample Size:",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 123,
            },
            "ref": null,
            "rendered": 123,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              Standard Mean:
            </div>,
            <div>
              1.23
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Standard Mean:",
              "className": "label",
            },
            "ref": null,
            "rendered": "Standard Mean:",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1.23,
            },
            "ref": null,
            "rendered": 1.23,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              Standard Deviation:
            </div>,
            <div>
              -1.5
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Standard Deviation:",
              "className": "label",
            },
            "ref": null,
            "rendered": "Standard Deviation:",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": -1.5,
            },
            "ref": null,
            "rendered": -1.5,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              Hypothesized Mean
            </div>,
            <div>
              2.34
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hypothesized Mean",
              "className": "label",
            },
            "ref": null,
            "rendered": "Hypothesized Mean",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 2.34,
            },
            "ref": null,
            "rendered": 2.34,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            onClick={[MockFunction]}
          >
            Back
          </button>,
          "className": "buttons",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Back",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <div
              className="label"
            >
              Sample Size:
            </div>
            <div>
              123
            </div>
          </div>,
          <div>
            <div
              className="label"
            >
              Standard Mean:
            </div>
            <div>
              1.23
            </div>
          </div>,
          <div>
            <div
              className="label"
            >
              Standard Deviation:
            </div>
            <div>
              -1.5
            </div>
          </div>,
          <div>
            <div
              className="label"
            >
              Hypothesized Mean
            </div>
            <div>
              2.34
            </div>
          </div>,
          <div
            className="buttons"
          >
            <button
              onClick={[MockFunction]}
            >
              Back
            </button>
          </div>,
        ],
        "className": "appValues",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                Sample Size:
              </div>,
              <div>
                123
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sample Size:",
                "className": "label",
              },
              "ref": null,
              "rendered": "Sample Size:",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 123,
              },
              "ref": null,
              "rendered": 123,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                Standard Mean:
              </div>,
              <div>
                1.23
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Standard Mean:",
                "className": "label",
              },
              "ref": null,
              "rendered": "Standard Mean:",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1.23,
              },
              "ref": null,
              "rendered": 1.23,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                Standard Deviation:
              </div>,
              <div>
                -1.5
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Standard Deviation:",
                "className": "label",
              },
              "ref": null,
              "rendered": "Standard Deviation:",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": -1.5,
              },
              "ref": null,
              "rendered": -1.5,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                Hypothesized Mean
              </div>,
              <div>
                2.34
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hypothesized Mean",
                "className": "label",
              },
              "ref": null,
              "rendered": "Hypothesized Mean",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2.34,
              },
              "ref": null,
              "rendered": 2.34,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[MockFunction]}
            >
              Back
            </button>,
            "className": "buttons",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Back",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Back",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
